src/View/Helper/UrlHelper.php

```diff
@@ -174,7 +174,7 @@ public function assetUrl($path, array $options = [])
         if (!array_key_exists('plugin', $options) || $options['plugin'] !== false) {
             list($plugin, $path) = $this->_View->pluginSplit($path, false);
         }
-        if (!empty($options['pathPrefix']) && $path[0] !== '/') {
+        if (!empty($options['pathPrefix']) && ($path === null || $path[0] !== '/')) {
```

Comment:
With an empty string you have the same issue...
Why not using `substr((string)...` here, that is safer.